pkg/controller/cloud/node_controller.go

```diff
@@ -418,11 +418,17 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 					n.Spec.ProviderID = providerID
 				}
 			})
+		} else if err == cloudprovider.NotImplemented {
+			// if the cloud provider being used does not support provider IDs,
+			// we can safely continue since we will attempt to set node
+			// addresses given the node name in getNodeAddressesByProviderIDOrName
+			klog.Errorf("failed to set node provider id, continuing: %v", err)
 		} else {
-			// we should attempt to set providerID on node, but
-			// we can continue if we fail since we will attempt to set
-			// node addresses given the node name in getNodeAddressesByProviderIDOrName
+			// if the cloud provider being used supports provider IDs, we want
+			// to propagate the error so that we re-try in the future; if we
+			// do not, the taint will be removed, and this will not be retried
 			klog.Errorf("failed to set node provider id: %v", err)
```

Comment:
I don't think we need to log error here since we're returning error.

Reply:
Makes sense. Removed!